src/node_report_module.cc

```diff
@@ -0,0 +1,315 @@
+  String::Utf8Value value(isolate, str);
+  TriggerNodeReport(
+      isolate,
+      env,
+      *value,
+      __func__,
+      nullptr,
+      MaybeLocal<Value>());
+}
+
+/*******************************************************************************
+ * Native module initializer function, called when the module is require'd
+ ******************************************************************************/
+void InitializeNodeReport(Isolate* isolate, Environment* env) {
+  // Register the boot time of the process, for
+  // computing resource consumption average etc.
+  // std::string options;
+  IsolateData* id = env->isolate_data();
+  std::shared_ptr<PerIsolateOptions> options = env->isolate_data()->options();
+  time(&boot_time);
+  if (options->report_catch_fatalerror && id->error_hook_initialised == false) {
+    env->isolate()->SetFatalErrorHandler(OnFatalError);
+    id->error_hook_initialised = true;
+  }
+
+  if (options->report_signal == "") options->report_signal = "SIGUSR2";
+}
+// A method to sync up data elements in the JS land with its
+// corresponding elements in the C++ world. Required because
+// (i) the tunables are first intercepted through the CLI but
+// later modified via APIs. (ii) the report generation events
+// are controlled partly from C++ and partly from JS.
+void SyncConfig(const FunctionCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  Local<Context> context = env->context();
+  std::shared_ptr<PerIsolateOptions> options = env->isolate_data()->options();
+  IsolateData* id = env->isolate_data();
+
+  CHECK_EQ(info.Length(), 2);
+  Local<Object> obj = info[0]->ToObject(context).ToLocalChecked();
+  bool sync = info[1].As<Boolean>()->Value();
+
+  // events array
+  Local<String> eventskey = FIXED_ONE_BYTE_STRING(env->isolate(), "events");
+  Local<Value> events_unchecked =
+      obj->Get(context, eventskey).ToLocalChecked();
+  Local<Array> events;
+  if (events_unchecked->IsUndefined() || events_unchecked->IsNull()) {
+    events_unchecked = Array::New(env->isolate(), 0);
+    obj->Set(context, eventskey, events_unchecked);
+  }
+  events = events_unchecked.As<Array>();
+
+  // signal
+  Local<String> signalkey = FIXED_ONE_BYTE_STRING(env->isolate(), "signal");
+  Local<Value> signal_unchecked =
+      obj->Get(context, signalkey).ToLocalChecked();
+  Local<String> signal;
+  if (signal_unchecked->IsUndefined() || signal_unchecked->IsNull())
+    signal_unchecked = FIXED_ONE_BYTE_STRING(env->isolate(), "signal");
+  signal = signal_unchecked.As<String>();
+
+  Utf8Value signalstr(env->isolate(), signal);
+
+  // report file
+  Local<String> filekey = FIXED_ONE_BYTE_STRING(env->isolate(), "filename");
+  Local<Value> file_unchecked = obj->Get(context, filekey).ToLocalChecked();
+  Local<String> file;
+  if (file_unchecked->IsUndefined() || file_unchecked->IsNull())
+    file_unchecked = FIXED_ONE_BYTE_STRING(env->isolate(), "filename");
```

Comment:
It’s definitely okay to use `FIXED_ONE_BYTE_STRING` for strings that are not listed in `env.h`, but you can just re-use `filekey` here, right?

Reply:
fixed